### DISCUSSION TITLE
Fit boundary of the 3D object with the width of the viewer window object

### DISCUSSION THREAD

**Discussion by jairovskaya** [google/model-viewer#4439] [Q&A] [answered]
Hi. Is there a way to make the boundary of the 3D object to fit to the max width of the viewer window object in html? Or: the camera fustrum to fit the boundary of the 3D object. the idea is to make the 3D object always to touch both sides of the window boundary.

**Comment by diegoteran** [comment]
Hi,
ModelViewer renders the 3D object such that any rotation fits in the viewer. 
Can you share some examples so we can understand better what you are looking for?

**Reply by jairovskaya** [reply]
Attached an image of the case. How to make the camera to exactly fit to the max width of the object boundary. I want to use a cylinder as a background element of a webpage so I need to fill out the whole space. Probably Model-Viewer is more intended for "objects" and not for backdrops; still, according to the API documentation seems that is possible to play around with the orbit-sensitivity to achieve the effect of observing from inside of a cave. I couldn't figure it out yet how though. 
![image](https://github.com/google/model-viewer/assets/50005193/dc5b955d-5329-4754-b5da-659a8493524b)

**Reply by diegoteran** [reply]
You might be looking for field of view: https://modelviewer.dev/docs/index.html#entrydocs-stagingandcameras-attributes-fieldOfView

**Reply by elalish** [reply]
The line you draw is not what I would consider the object boundary (that would be the max width, lower down). You can set [`camera-orbit`](https://modelviewer.dev/docs/#entrydocs-stagingandcameras-attributes-cameraOrbit) to `"auto auto 100%"` to remove our default 5% margin, but if your object is asymmetric, you'll likely still have margin on one side or the other as you rotate.

**Comment by jairovskaya** [comment]
Understood. I tried it but didn't get me the result I'm looking for. is there a way to make the camera to be in the middle of the object? So I can see the object from inside (cave effect).

**Reply by elalish** [reply, marked as answer]
For the inside you set `min-camera-orbit` and `camera-orbit` to `"auto auto 0m"`. Look through to docs for other useful configurations, like setting `orbit-sensitivity` to -1: https://modelviewer.dev/docs/#entrydocs-stagingandcameras-attributes-orbitSensitivity

**Reply by jairovskaya** [reply]
Woooo! Superb. This is what I was looking for. Thank you!. You guys should add a sample of this type of view, useful to add environments for the objects.

**Reply by elalish** [reply]
A sample would be great! We tend to be model-limited around here (we're 3D devs, not artists). We always welcome PRs with additional examples, and we also use them as manual test cases, so if there's a feature you use and want to make sure we don't break, that's a great way to protect yourself.
